fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -316,6 +317,13 @@ public static <T extends BinaryOperator> T processCharacterLiteralInBinaryOperat
     @Developing
     public static Optional<Expression> characterLiteralTypeCoercion(String value, DataType dataType) {
         Expression ret = null;
+        if (Config.enable_date_conversion) {
+            if (dataType instanceof DateType) {
+                dataType = DateV2Type.INSTANCE;
+            } else if (dataType instanceof DateTimeType) {
+                dataType = DateTimeV2Type.MAX;
+            }
+        }
```

Comment:
why do this?

Reply:
because we will transform the datev1 type to datev2 type if enable date conversion.